tephra-hbase-compat-1.1-base/src/main/java/org/apache/tephra/hbase/txprune/PruneUpperBoundWriter.java

```diff
@@ -81,14 +84,23 @@ private void startFlushThread() {
     flushThread = new Thread("tephra-prune-upper-bound-writer") {
       @Override
       public void run() {
-        while ((!isInterrupted()) && isRunning()) {
+        Service.State serviceState = state();
+        while ((!isInterrupted()) &&
+          (serviceState.equals(Service.State.NEW) || serviceState.equals(Service.State.STARTING) ||
```

Comment:
hmm - do you think it would simplify the code if we go back to using a `stop` flag?